draft-irtf-cfrg-opaque.md

```diff
@@ -371,20 +371,32 @@ OPAQUE relies on the following protocols and primitives:
 Note that we only need the base mode variant (as opposed to the verifiable mode
 variant) of the OPRF described in {{I-D.irtf-cfrg-voprf}}.
 
-- Cryptographic hash function:
-  - Hash(m): Compute the cryptographic hash of input message `m`. The type of the
-    hash is determined by the chosen OPRF group.
-  - Nh: The output size of the Hash function.
-
-- Authenticated Key Exchange (AKE, {{instantiations}}):
-  - Npk: The size of the public keys used for the key exchange protocol.
-  - Nsk: The size of the private keys used for the key exchange protocol.
+- Key Derivation Function (KDF):
+  - Extract(salt, ikm): Extract a pseudorandom key of fixed length `Nl` bytes from
+    input keying material `ikm` and an optional byte string `salt`.
+  - Expand(prk, info, L): Expand a pseudorandom key `prk` using optional string `info`
+    into `L` bytes of output keying material.
+  - Nl: The output size of the `Extract()` function in bytes.
+
+- Message Authentication Code (MAC):
+  - MAC(key, msg): Compute a message authentication code over input `msg` with key
+    `key`, producing a fixed-length output of `Nm` bytes.
+  - Nm: The output size of the `MAC()` function in bytes.
+
+- Hash Function:
+  - Hash(msg): Apply a cryptographic hash function to input `msg`, producing an
+    fixed-length digest of size `Nh` bytes.
+  - Nh: The output size of the `Hash()` function in bytes.
 
 - Memory Hard Function (MHF):
   - Harden(msg, params): Repeatedly apply a memory-hard function with parameters
     `params` to strengthen the input `msg` against offline dictionary attacks.
     This function also needs to satisfy collision resistance.
 
+- Authenticated Key Exchange (AKE, {{instantiations}}):
+  - Npk: The size of the public keys used for the key exchange protocol.
+  - Nsk: The size of the private keys used for the key exchange protocol.
+
```

Comment:
It can make sense to specify the necessary Group operations in 3DH.
A "3DH Group" (or so) block could outline the necessary functions (e.g. KeyGen() for the ephemeral keys (or _GetRandomScalar()_ and _ScalarBaseMult()_ ) and ScalarMult() ).

Reply:
This would be 3DH-specific, though. Also, presumably we would reuse the corresponding functions in the OPRF bullet point for 3DH.

Reply:
See above -- I'll just say we assume a prime-order group. That will help your internal/external PR, too, since I'll make HashToScalar a part of that interface.